pkgs/development/libraries/alkimia/default.nix

```
@@ -14,7 +14,7 @@ mkDerivation rec {
 
   nativeBuildInputs = [ extra-cmake-modules doxygen graphviz ];
 
-  buildInputs = [ qtbase qtwebkit kdelibs4support plasma-framework knewstuff kpackage ];
+  buildInputs = [ qtbase qtwebengine kdelibs4support plasma-framework knewstuff kpackage ];
```

`alkemia` seems to build even if neither `qtwebkit` nor `qtwebengine` are provided, but looking at the code, that (i.e. absence of both) seems to disable some features. Whether or not these features are needed for KMyMoney or other dependents, I don't know, thus why I didn't remove `qtwebkit` without replacement and instead replaced it by `qtwebengine`. (I hope that CMake automatically picks that up. Should probably check in the build log.)

---

> `alkemia` seems to build even if neither `qtwebkit` nor `qtwebengine` are provided, but looking at the code, that (i.e. absence of both) seems to disable some features. Whether or not these features are needed for KMyMoney or other dependents, I don't know, thus why I didn't remove `qtwebkit` without replacement and instead replaced it by `qtwebengine`. (I hope that CMake automatically picks that up. Should probably check in the build log.)

This could be written as a comment IMO.

---

> This could be written as a comment IMO.

Do you mean as a code comment in the source file itself, rather than just a comment on the PR? Like so:?
```suggestion
  # qtwebengine is not needed for building,
  # but when both qtwebengine and qtwebkit would be absent at build-time,
  # some features of alkemia would be disabled would be unavailable.
  # As we are unsure whether those features are needed,
  # we provide qtwebengine here.
  buildInputs = [ qtbase qtwebengine kdelibs4support plasma-framework knewstuff kpackage ];
```

---

Maybe something shorter:

```suggestion
  # qtwebengine is not a mandatory dependency, but it adds some features we need for
  # alkimia's dependents. See:
  # https://github.com/KDE/alkimia/blob/v8.1.1/CMakeLists.txt#L124
  buildInputs = [ qtbase qtwebengine kdelibs4support plasma-framework knewstuff kpackage ];
```

---

Well, as written, I'm not _sure_ those features are being used / needed. (Nor that they're _not_ used / needed.) So maybe
```suggestion
  # qtwebengine is not a mandatory dependency, but it adds some features
  # we might need for alkimia's dependents. See:
  # https://github.com/KDE/alkimia/blob/v8.1.1/CMakeLists.txt#L124
  buildInputs = [ qtbase qtwebengine kdelibs4support plasma-framework knewstuff kpackage ];
```

---

> Well, as written, I'm not _sure_ those features are being used / needed. (Nor that they're _not_ used / needed.)

I thought you concluded from testing kmymoney with alkamia built with and without qtwebengine that it needs it. In anycase, your suggestion is good as well.

---

I've just verified that KMyMoney builds and runs in both cases. I didn't notice any differences, but can't be sure there aren't any. (And as KMyMoney has its own direct dependency on qtwebengine anyway, I guess it doesn't matter too much that we let alkimia depend on qtwebengine, too.)

I've included the comment in the code now.

---

> I've included the comment in the code now.

I don't see the comment..

---

https://github.com/NixOS/nixpkgs/blob/8d9eab433a0c23f25a7bdeb9e56b6170b0eeb9fb/pkgs/development/libraries/alkimia/default.nix#L17-L20
[change to previously reviewed version](https://github.com/NixOS/nixpkgs/compare/0f7f6dcda585ca530e251ab68b17971c7bd76c32..8d9eab433a0c23f25a7bdeb9e56b6170b0eeb9fb)